HAProxy: update the recommended hardware configuration for HAProxy (#16674)
pingcap/docs: best-practices/haproxy-best-practices.md

### Hardware requirements

> According to the [HAProxy documentation](https://www.haproxy.com/documentation/haproxy-enterprise/getting-started/installation/linux/), the minimum hardware configuration for HAProxy is shown in the following table. Under the sysbench `oltp_read_write` workload, the maximum QPS for this configuration is about 50K. You can also improve server specifications according to the load balancing environment.

Author: ```suggestion
According to the [HAProxy documentation](https://www.haproxy.com/documentation/haproxy-enterprise/getting-started/installation/linux/), the minimum hardware configuration for HAProxy is shown in the following table. Under the Sysbench `oltp_read_write` workload, the maximum QPS for this configuration is about 50K. You can increase the server configuration according to your load balancing environment.
```

> According to the [HAProxy documentation](https://www.haproxy.com/documentation/haproxy-enterprise/getting-started/installation/linux/), the minimum hardware configuration for HAProxy is shown in the following table. Under the sysbench `oltp_read_write` workload, the maximum QPS for this configuration is about 50K. You can also improve server specifications according to the load balancing environment.

Reviewer: ```suggestion
According to the [HAProxy documentation](https://www.haproxy.com/documentation/haproxy-enterprise/getting-started/installation/linux/), the minimum hardware configuration for HAProxy is shown in the following table. Under the Sysbench `oltp_read_write` workload, the maximum QPS for this configuration is about 50K. You can increase the server configuration according to your load balancing environment.
```

> According to the [official documentation](https://www.haproxy.com/documentation/haproxy-enterprise/getting-started/installation/linux/), the minimum hardware configuration for HAProxy is shown in the following table. Under the sysbench `oltp_read_write` workload, the maximum QPS for this configuration is about 50K. You can also improve server specifications according to the load balancing environment.

Reviewer: ```suggestion
According to the [HAProxy documentation](https://www.haproxy.com/documentation/haproxy-enterprise/getting-started/installation/linux/), the minimum hardware configuration for HAProxy is shown in the following table. Under the Sysbench `oltp_read_write` workload, the maximum QPS for this configuration is about 50K. You can increase the server configuration according to your load balancing environment.
```
